test/parse/test_entity_declaration.rb

```diff
@@ -23,6 +23,158 @@ def parse(internal_subset)
     end
 
     public
+
+    class TestGarbagedValue < self
+      def test_garbaged_after_entity_definition
+        exception = assert_raise(REXML::ParseException) do
+          REXML::Document.new('<!DOCTYPE root [<!ENTITY "foo" xxx > ]>')
```

Comment:
Could you use more easy to understand sample values instead of `"foo"` and `xxx`?
For example, `"valid-name"` and `invalid-external-id`.